### PR TITLE
*: save kv.Storage in import disttask components and use it

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #51337

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > pass existing tests

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/importer/table_import.go [hunk 1]
```diff
@@ -214,12 +194,6 @@ func NewTableImporter(param *JobImportParam, e *LoadDataController, id string) (
...
 	}
 
-	// no need to close kvStore, since it's a cached store.
-	kvStore, err := GetCachedKVStoreFrom(tidbCfg.Path, tls)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-
 	backendConfig := e.getLocalBackendCfg(tidbCfg.Path, dir)
 	d := kvStore.(tikv.Storage).GetRegionCache().PDClient().GetServiceDiscovery()
```
Comment (D3Hunter) [inline pkg/executor/importer/table_import.go:198]:
```suggestion
	d := kvStore.(tidbkv.StorageWithPD).GetPDClient().GetServiceDiscovery()
```

File: pkg/disttask/importinto/scheduler.go [hunk 2]
```diff
@@ -136,6 +139,8 @@ type ImportSchedulerExt struct {
 	// It may be changed when we switch to a new task or switch to a new owner.
 	currTaskID            atomic.Int64
 	disableTiKVImportMode atomic.Bool
+
+	store kv.Storage
```
Comment (D3Hunter) [inline pkg/disttask/importinto/scheduler.go:143]:
```suggestion
	storeWithPD tidbkv.StorageWithPD
```
we don't need depends on a heavy `store`.

at task executor part we only need a tidbkv.StorageWithPD and a `keyspaceGetter`, but we can change it later.

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: pkg/disttask/importinto/scheduler.go [hunk 3]
```diff
@@ -403,13 +422,21 @@ func (sch *ImportSchedulerExt) updateCurrentTask(task *proto.Task) {
...
 }
 
-func newImportScheduler(ctx context.Context, task *proto.Task, param scheduler.Param) scheduler.Scheduler {
+// NewImportScheduler creates a new import scheduler.
+func NewImportScheduler(
+	ctx context.Context,
+	task *proto.Task,
+	param scheduler.Param,
+	store kv.Storage,
```
Comment (D3Hunter) [inline pkg/disttask/importinto/scheduler.go:433]: we can move it to `scheduler.Param` if `ADD-INDEX` need it later?

Comment (lance6716) [reply pkg/disttask/importinto/scheduler.go:433]: ADD INDEX will get the store from `ddl` object.

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**